siwe/siwe.py

```diff
@@ -210,10 +220,10 @@ def validate(
         try:
             address = w3.eth.account.recover_message(message, signature=signature)
         except eth_utils.exceptions.ValidationError:
-            raise InvalidSignature
+            raise InvalidSignature("Failed to recover public key from signature")
 
         if address != self.address:
-            raise InvalidSignature
+            raise InvalidSignature("Signer address must match message address")
```

Comment:
The message is misleading, the signature might have been created by the correct address but if the message changed then you will recover a different public key.

Reply:
that's a good point. would you accept a change to `Recovered address does not match message address`? 

FWIW I think that the Go repo's implementation should be updated then: https://github.com/spruceid/siwe-go/blob/main/siwe.go#L218

Reply:
@0xdaem0n one note is that some of the SIWE libraries are built by the community, so there may not be total standardization across them. IIRC, the Go implementation was by a community member in Discord. I built this library (with the help of @sbihel ) and I have no professional affiliation with SpruceID.

In terms of standardizing library behavior, I think https://github.com/spruceid/siwe is the best source of truth.